### PR TITLE
docs: mutation-testing

### PR DESCRIPTION
**Pull request by ASuciuX**
Added mutation testing-related info:
- testing docs file
  - how to run mutation testing locally
- mutation testing check on the PR template
- contributing section for how mutation testing should be taken into account when contributing

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 ### Checklist
 
 - [ ] Test coverage for new or modified code paths
+- [ ] Mutation testing passing for new or modified functions
```
Comment (wileyj) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: can you add a link to the new testing doc here?
in fact, i think would be a good idea to link to he relevant locations of that doc for each testing step

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+### Overview
+
+Mutation testing is complementary to code coverage.
+It evaluates the tests' quality of the functions added or modified in the PR.
+Mutation testing involves making small changes (mutations) to the code to check if the tests can detect these changes.
+This mutations represent a clone of the original code with one given function having a default value instead of its body.
+Then the `cargo build` and `cargo nextest` for the package intended are called. If the mutated code breaks, this means that function is properly tested.
+
+This is useful for any size of changed code, being it one line in one function, or tens and hundreds of functions, but the time required increases directly with the number of mutants.
```
Comment (wileyj) [inline docs/testing.md:19]: I would prefer to keep this simple - perhaps a link to wikipedia etc about what mutation testing is?

**Review by wileyj — APPROVED** [review]
minor changes. but looks good

**Comment by ASuciuX** [comment]
> understood! i'll remove my approval for now - can you convert this to draft since we're waiting on other work?
Yes, I'll convert it now.

**Comment by ASuciuX** [comment]
This is outdated atm, would prefer to open a new one when it will be the case.

**Pull request closed without merging**